app/file.go

```diff
@@ -1129,6 +1129,15 @@ func (a *App) generateMiniPreview(fi *model.FileInfo) {
 			mlog.Debug("generateMiniPreview: prepareImage failed", mlog.Err(imgErr),
 				mlog.String("fileinfo_id", fi.Id), mlog.String("channel_id", fi.ChannelId),
 				mlog.String("creator_id", fi.CreatorId))
+
+			// Since this file is not a valid image (for whatever reason), prevent this fileInfo
+			// from entering generateMiniPreview in the future
+			fi.UpdateAt = model.GetMillis()
+			fi.MimeType = "invalid-" + fi.MimeType
+			if _, err2 := a.Srv().Store.FileInfo().Upsert(fi); err2 != nil {
+				mlog.Debug("FileInfo save failed", mlog.Err(err2))
+			}
```

Comment:
nit: Propose avoiding `err2` by writing as (plus tweak to error text):
```suggestion
			err = a.Srv().Store.FileInfo().Upsert(fi); 
			if err != nil {
				mlog.Debug("Invalidating FileInfo failed", mlog.Err(err))
			}
```

Reply:
Fixed. :)